src/restricted-endpoints/replication/changes/changes.controller.ts

```diff
@@ -0,0 +1,61 @@
+import { Controller, Get, Param, Query, UseGuards } from '@nestjs/common';
+import { ChangesResponse } from '../bulk-document/couchdb-dtos/changes.dto';
+import { CouchdbService } from '../../../couchdb/couchdb.service';
+import { map, Observable } from 'rxjs';
+import { OnlyAuthenticated } from '../../../auth/only-authenticated.decorator';
+import { CombinedAuthGuard } from '../../../auth/guards/combined-auth/combined-auth.guard';
+import { User } from '../../../auth/user.decorator';
+import { UserInfo } from '../../session/user-auth.dto';
+import {
+  DocumentAbility,
+  PermissionService,
+} from '../../../permissions/permission/permission.service';
+import { DatabaseDocument } from '../bulk-document/couchdb-dtos/bulk-docs.dto';
+import { omit } from 'lodash';
+
+@OnlyAuthenticated()
+@UseGuards(CombinedAuthGuard)
+@Controller()
+export class ChangesController {
+  constructor(
+    private couchdbService: CouchdbService,
+    private permissionService: PermissionService,
+  ) {}
+
+  /**
+   * Get the changes stream.
+   * The changes feed only returns the doc IDs to which the requesting user has access.
+   * Even if `include_docs: true` is set, the stream will not return the document content.
+   * @param db
+   * @param params
+   * @param user
+   */
+  @Get(':db/_changes')
+  changes(
+    @Param('db') db: string,
+    @Query() params,
+    @User() user: UserInfo,
+  ): Observable<ChangesResponse> {
+    return this.couchdbService
+      .get<ChangesResponse>(db, '_changes', {
+        ...params,
+        include_docs: true,
+      })
+      .pipe(map((res) => this.filterChanges(res, user)));
+  }
+
+  private filterChanges(
+    changes: ChangesResponse,
+    user: UserInfo,
+  ): ChangesResponse {
+    const ability = this.permissionService.getAbilityFor(user);
+    changes.results = changes.results
+      .filter((change) => this.canReadDoc(change.doc, ability))
+      .map((change) => omit(change, 'doc'));
```

Comment:
We could now enable the `include_docs: true` option again by not omitting the `doc` property in that case. But for the normal usage this does not add much benefit.

Reply:
hmm, I guess permissions are ensured now either way :-) 
Shall we just leave it to the client then to request also with docs, if it wants?

Reply:
_deleted docs can not be requested with full body using _include_docs, can they?

Reply:
Depends a little. The way we normally do it, the document content is removed. But it is also possible to only delete a docs by setting `_deleted: true` manually and saving it. In this case the document content is kept and it would be returned here. 
The problem is, it is not possible to distinguish these cases. As we are only working with the former way I think this is a ok limitation.

Reply:
if the doc content might be included, we shouldn't permit include_docs on changes. But that's alright I think.

Reply:
Ok so you suggest to always remove the `doc` property?